pvlib/snow.py

```diff
@@ -185,3 +185,130 @@ def dc_loss_nrel(snow_coverage, num_strings):
        Available at https://www.nrel.gov/docs/fy18osti/67399.pdf
     '''
     return np.ceil(snow_coverage * num_strings) / num_strings
+
+
+def _townsend_effective_snow(snow_load, snow_events):
+    '''
+    Calculates effective snow using the total snowfall in inches received
+    each month and the number of snowfall events each month.
+
+    Parameters
+    ----------
+    snow_load : array-like
+        Inches of snow received each month. Referred to as S in the paper [in]
+
+    snow_events : array-like
+        Number of snowfall events each month. Referred to as N in the paper [-]
+
+    Returns
+    -------
+    effective_snowfall : array-like
+        Effective snowfall as defined in the Townsend model
+
+    References
+    ----------
+    .. [1] Townsend, Tim & Powers, Loren. (2011). Photovoltaics and snow: An
+       update from two winters of measurements in the SIERRA. Conference
+       Record of the IEEE Photovoltaic Specialists Conference.
+       003231-003236. :doi:`10.1109/PVSC.2011.6186627`
+       Available at https://www.researchgate.net/publication/261042016_Photovoltaics_and_snow_An_update_from_two_winters_of_measurements_in_the_SIERRA
+    '''  # noqa: E501
+    snow_events_no_zeros = np.maximum(snow_events, 1)
+    effective_snow = 0.5 * snow_load * (1 + 1 / snow_events_no_zeros)
+    return np.where(snow_events > 0, effective_snow, 0)
+
+
+def loss_townsend(snow_total, snow_events, surface_tilt, relative_humidity,
+                  temp_air, poa_global, slant_height, lower_edge_drop_height,
+                  angle_of_repose=40):
+    '''
+    Calculates monthly snow loss based on the Townsend monthly snow loss model [1]_.
+
+    Parameters
+    ----------
+    snow_total : array-like
+        Inches of snow received each month. Referred to as S in the paper [in]
```

Comment:
@reepoi note that I think changing to metric requires more than just editing the docstring -- I think some conversions will be needed in the code as well, since the equations and coefficients in the reference assume non-metric inputs.

Reply:
Yes, the code does need to be changed, and here is a description of what I did. Also, I want to bring attention to the variable `loss_percentage` ([this line](https://github.com/pvlib/pvlib-python/blob/bbe5bc33f8e6fff0d0e405484a7bd70863cc6c45/pvlib/snow.py#L305)) and how it is affected by the unit conversion. I'll describe it more next.
Code changes for converting to metric:
- Method `_townsend_effective_snow`: This method's code was not changed. Here is the line calculates effective snow: `effective_snow = 0.5 * snow_total * (1 + 1 / snow_events_no_zeros)`. Let `X = 0.5 * (1 + 1 / snow_events_no_zeros)`, which is unitless. `effective_snow = X * snow_total` and `effective_snow` has the same units as `snow_total`.
- Method `loss_townsend`: This method's code was changed. The variable `gamma` is unitless in the paper, so it should remain unitless after changing to metric. Here is the formula for `gamma`:
  ```slant_height (in) * effective_snow_weighted (in) / (((drop_height_clipped (in))^2 - (effective_snow_weighted (in))^2) * 2 * tand(angle_of_repose)```
The (in)^2 cancel. Next change the inputs to metric. `effective_snow_weighted` is now in centimeters. Let `effective_snow_weighted_m` be `effective_snow_weighted` in meters. The formula for `gamma` becomes
```slant_height (m) * effective_snow_weighted_m (m) / (((drop_height_clipped (m))^2 - (effective_snow_weighted (m))^2) * 2 * tand(angle_of_repose)```
The (m)^2 cancel.
I compared the values of `gamma` calculated using the unit test data ([from this test](https://github.com/pvlib/pvlib-python/blob/bbe5bc33f8e6fff0d0e405484a7bd70863cc6c45/pvlib/tests/test_snow.py#L110)) in imperial units and in metric units, and they were the same. Note the data of the test I linked is in metric.
`ground_interference_term` is also unitless.

Next, changing `loss_percentage` to metric affects the result of `loss_townsend` because it is not unitless. The unit of the returned result `Loss` in the Townsend-Powers paper is `inches/(Kelvin)^2/(kWh/m2)^0.67`. From the unit conversions to metric (and changing `poa_global` from kWh/m2 to Wh/m2), `loss_percentage` has the unit `centimeters/(Kelvin)^2/(Wh/m2)^0.67`. Does either of these units make sense as a capacity loss fraction?

Reply:
Eq.3 gives the loss percentage if I am reading this paper correctly. I would assume that the constant C1 has units that, when multiplied by the other factors, give a product with unit kWhr_DC (loss) / kWhr_DC (no snow) * 100%. C1 is the slope of the fitted line in Fig. 7, so the unit is the y-axis unit (kWhr (loss) / kWhr (no snow) * 100) divided by the bizarre unit on the x-axis.

Bottom line: I would convert `poa_global` from its input unit of Whr /m2 to kWh / m2 inside the function, and not change C1's value.

Reply:
Ah, I see. Since weighted effective snow (inches) is also in the unit of Figure 7's x-axis, we should also convert `effective_snow_weighted` from centimeters to inches in the `loss_percentage` calculation.